### DISCUSSION TITLE
Can i create element of model without id?

### DISCUSSION THREAD

**Discussion by verylovestars** [strapi/strapi#7403] [Legacy Topics]
Hello! I ran into a problem, the solution of which seems difficult to me. I have a model in the DB that does not have an id column. Instead, a composite primary key is used with 3 elements. So when I try to create a new item, I get an error.

![Primary Key](https://user-images.githubusercontent.com/39151330/89795673-5722cb80-db31-11ea-96dd-9dd47ed512da.png)

![An Error](https://user-images.githubusercontent.com/39151330/89795739-6d308c00-db31-11ea-85ed-447fa4d1fea2.png)

**Comment by derrickmehaffy** [comment]
Models currently require a default `id` as a primary key for the moment and we don't currently allow changing the id name or type. This is something we are looking into for future versions but currently the primary key `id` is hard coded in much of the backend.

See the following FAQ article on our docs: https://strapi.io/documentation/v3.x/getting-started/troubleshooting.html#does-strapi-allow-me-to-change-the-default-id-type-or-name

Unfortunately this is not a simple issue to fix locally within a single project without forking and maintaining multiple packages.
